src/resolvers/user_management.spec.js

```diff
@@ -21,59 +21,71 @@ const factory = Factory()
+const jennyRostocksHeaders = {
+  authorization:
+    "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJyb2xlIjoidXNlciIsImxvY2F0aW9uTmFtZSI6bnVsbCwibmFtZSI6Ikplbm55IFJvc3RvY2siLCJhYm91dCI6bnVsbCwiYXZhdGFyIjoiaHR0cHM6Ly9zMy5hbWF6b25hd3MuY29tL3VpZmFjZXMvZmFjZXMvdHdpdHRlci9zYXNoYV9zaGVzdGFrb3YvMTI4LmpwZyIsImlkIjoidTMiLCJlbWFpbCI6InVzZXJAZXhhbXBsZS5vcmciLCJzbHVnIjoiamVubnktcm9zdG9jayIsImlhdCI6MTU1MDg0NjY4MCwiZXhwIjoxNjM3MjQ2NjgwLCJhdWQiOiJodHRwOi8vbG9jYWxob3N0OjMwMDAiLCJpc3MiOiJodHRwOi8vbG9jYWxob3N0OjQwMDAiLCJzdWIiOiJ1MyJ9.eZ_mVKas4Wzoc_JrQTEWXyRn7eY64cdIg4vqQ-F_7Jc"
+};
 
 beforeEach(async () => {
-  await factory.create('User', {
-    avatar: 'https://s3.amazonaws.com/uifaces/faces/twitter/seyedhossein1/128.jpg',
-    id: 'acb2d923-f3af-479e-9f00-61b12e864666',
-    name: 'Matilde Hermiston',
-    slug: 'matilde-hermiston',
-    role: 'user',
-    email: 'test@example.org',
-    password: '1234'
-  })
-})
+  await factory.create("User", {
+    avatar:
+      "https://s3.amazonaws.com/uifaces/faces/twitter/seyedhossein1/128.jpg",
+    id: "acb2d923-f3af-479e-9f00-61b12e864666",
+    name: "Matilde Hermiston",
+    slug: "matilde-hermiston",
+    role: "user",
+    email: "test@example.org",
+    password: "1234"
+  });
+});
 
 afterEach(async () => {
-  await factory.cleanDatabase()
-})
-
-describe('isLoggedIn', () => {
-  const query = '{ isLoggedIn }'
-  describe('unauthenticated', () => {
-    it('returns false', async () => {
-      await expect(request(host, query)).resolves.toEqual({ isLoggedIn: false })
-    })
-  })
-
-  describe('with malformed JWT Bearer token', () => {
-    const headers = { authorization: 'blah' }
-    const client = new GraphQLClient(host, { headers })
-
-    it('returns false', async () => {
-      await expect(client.request(query)).resolves.toEqual({ isLoggedIn: false })
-    })
-  })
-
-  describe('with valid JWT Bearer token', () => {
-    const client = new GraphQLClient(host, { headers: jennyRostocksHeaders })
-
-    it('returns false', async () => {
-      await expect(client.request(query)).resolves.toEqual({ isLoggedIn: false })
-    })
-
-    describe('and a corresponding user in the database', () => {
-      it('returns true', async () => {
+  await factory.cleanDatabase();
+});
+
+describe("isLoggedIn", () => {
+  const query = "{ isLoggedIn }";
+  describe("unauthenticated", () => {
+    it("returns false", async () => {
+      await expect(request(host, query)).resolves.toEqual({
```

Comment:
I found it a possible error cause for the testing to use `request` when I would use `new GrapqhQLClient` further down in the test. I like to put everything that changes in a `beforeEach` block and limit the relevant code to a few lines. So you could e.g. initialize `headers = {}` in `beforeEach` and in the authenticated block you override `headers = await login(...)` in a `beforeEach`.

This saved me some debugging after doing copy+paste.

Reply:
Not sure I get you here.
it doesn't use headers not sure why you want me to add?

Reply:
It's basically the same suggestion as below: By re-using the very same set of actions you can avoid errors in the test setup. Only a few parameters are iterated in the tests and so you can make sure, it's the *parameters* that make the difference, not the slightly different test-setup. (e.g. `request` instead of using a `GraphQLClient`).